src/types/records-types.ts

```diff
@@ -203,10 +203,17 @@ export type RecordsReadMessage = {
 };
 
 export type RecordsReadReply = GenericMessageReply & {
+  /**
+   * The RecordsDelete if the record is deleted.
+   */
+  delete?: RecordsDeleteMessage;
+
+  /**
+   * The initial write of the record if the returned RecordsWrite message itself is not the initial write or if a RecordsDelete is returned.
+   */
+  initialWrite?: RecordsWriteMessage;
+
```

Comment:
This serves as a starting point for discussion and needs to be finalized. While this approach is backwards compatible, it is but far from ideal. I personally would lean towards just revamping this type entirely and just require the downstream dependency (mainly `web5-js`) update its code. A few things to NOTE:

1. I considered overloading/polluting existing property `record` further, but `data` within it is currently required.
1. `initialWrite` is now duplicated, I am considering to take it out as a top level property (as shown in PR).
1. the naming `record` is less than ideal: it makes the peer property `delete` look out of place. Perhaps we should name them `recordsWrite` and `recordsDelete` or just `write` and `delete` respectively? Or we if combine the two message types (pt. 1), then maybe we call it `state` or `message`?!

Reply:
Yeah I agree that the current form isn't ideal.

Curious to see what combining the two message types could look like, that might be more consistent with how it's normally consumed (just a record and a it's current state).

Reply:
Decided to submit a separate PR because it actually touches a bunch of files, so don't want this change to be drowned out by the refactor noise.